### ISSUE TITLE
(path/parse) suggestion: ParsedPath.toString return string representing full path

### ISSUE THREAD

**Issue by zachauten** [denoland/deno_std#2498]
**Is your feature request related to a problem? Please describe.**

It would be nice if [ParsedPath.toString](https://deno.land/std@0.151.0/path/mod.ts?s=ParsedPath) returned a string of the path that created it, like [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/toString). Currently it returns `"[object Object]"`.
Example:
```
> const p = parse("/usr/local/bin")
> p.toString()
"/usr/local/bin"
```

**Describe alternatives you've considered**
Adding a property to ParsedPath that represents the full path, like URL.href.

Or just leave it as is. You can achieve the same thing with `parsed.dir + "/" + parsed.base`, `join(parsed.dir, parsed.base)`, or just saving the string you pass into `parse`.

**Comment by lino-levan** [comment]
@iuioiua thoughts on this one? It feels a little more java/python-esque than javascript but if I squint a little bit I could see how this is useful. I'm personally leaning against it but I don't have a strong opinion either way.

**Comment by iuioiua** [comment]
Seeing as this suggestion hasn't gotten much attention or support from the community, it may not be essential, per se. Nevertheless, I'm slightly +1 for it. Being able to convert an object into a string via a `.toString()` method is a common and useful pattern.

**Comment by timreichen** [comment]
I would like to throw in the idea of replacing the bare object implementation with a `Path` class.

**Comment by jsejcksn** [comment]
> Being able to convert an object into a string via a `.toString()` method is a common and useful pattern.

I support this standard approach.

> I would like to throw in the idea of replacing the bare object implementation with a `Path` class.

This would be useful for mitigating some pitfalls of the current design — for example: modifying the `ext` property of a `ParsedPath` will have no effect on the `base` property, which also means that it won't have any effect when formatting the path. Consider this example which demonstrates what I described:

```ts
import { format } from "https://deno.land/std@0.223.0/path/posix/format.ts";
import { parse } from "https://deno.land/std@0.223.0/path/posix/parse.ts";

const parsed = parse("path/to/file.png");

console.log({
  base: parsed.base, // "file.png"
  ext: parsed.ext, // ".png"
});

parsed.ext = ".jpg";

const actual = format(parsed);
const expected = "path/to/file.jpg";

console.log(actual === expected, actual); // false "path/to/file.png"
console.log(parsed.base === "file.jpg", parsed.base); // false "file.png"

```

By using a class with getters and setters, the relationships of these properties can be properly maintained by recomputing them during mutations.

**Comment by kt3k** [comment]
I'm personally open to the Path class idea.

There was also past discussions (though it's rejected in them):
- https://github.com/denoland/deno_std/issues/2072
- https://github.com/denoland/deno_std/issues/678

There's also a PR to implement [dax](https://github.com/dsherret/dax)-inspired Path class
- https://github.com/denoland/deno_std/pull/4250

**Comment by iuioiua** [comment]
Yeah, I might like it if there was a `Path` class either, as long as we kept the API simple and inspired mainly by the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) interface, which I like.

**Comment by iuioiua** [comment]
In #4250, I think `Path` does far too much. For example, I don't think it should not use any `Deno.FsFile` methods.
